### PR TITLE
Feature library backend

### PR DESCRIPTION
**Pull request by Roy-Graafmans**
The track constructor is now dynamic with a option to add additional information from another pre loaded file

### CODE REVIEW

File: contextproject-ms3/src/main/java/contextproject/models/Track.java [hunk 1]
```diff
@@ -29,7 +31,10 @@
    * @param abPath
    *          Path of the mp3 file
    */
-  public Track(String abPath) {
+  public Track(String abPath, Hashtable<String, String> info) {
```
Comment (daveystruijk) [inline contextproject-ms3/src/main/java/contextproject/models/Track.java]: Could you leave the old Track constructor (with only abPath) as well? (and then let them both call the same method). Then everything will be nice and backwards compatible.

File: contextproject-ms3/src/main/java/contextproject/models/Track.java [hunk 2]
```diff
@@ -29,7 +31,10 @@
...
+
+    // String title, String artist, String album, long length, double bpm, String key, long
+    // firstBeat, int beatIntroStart, int beatsIntroLength, int beatOutroStart, int beatsOutroLength
```
Comment (daveystruijk) [inline contextproject-ms3/src/main/java/contextproject/models/Track.java]: Would be nice if you could update the constructor comment with this information.

File: contextproject-ms3/src/main/java/contextproject/models/Track.java [hunk 3]
```diff
@@ -42,8 +47,60 @@ public Track(String abPath) {
...
+    try {
+      firstBeat = Long.parseLong(info.get("firstBeat"));
+    } catch (NumberFormatException e) {
+      firstBeat = 0;
+    }
+
+    this.beatGrid = new BeatGrid(this.length, this.bpm, firstBeat, startBeatIntro, introBeatLength,
+        startBeatOurto, outroBeatLength);
+
   }
```
Comment (Emiel94) [inline contextproject-ms3/src/main/java/contextproject/models/Track.java]: Isn't it better to call a method in the constructor that does all the work? It looks pretty messy to have a constructor with that many lines of code.

File: contextproject-ms3/src/test/java/contextproject/models/BeatGridTest.java [hunk 4]
```diff
@@ -8,10 +8,19 @@
 public class BeatGridTest {
 
   private BeatGrid beatGrid;
+  private BeatGrid beatGrid2;
+  private BeatGrid beatGrid3;
 
+  /**
+   * @Before.
+   */
```
Comment (Emiel94) [inline contextproject-ms3/src/test/java/contextproject/models/BeatGridTest.java]: That's a weird javadoc?

**Pull request merged**